python/ray/ml/preprocessor.py

```diff
@@ -89,6 +89,20 @@ def fit(self, dataset: Dataset) -> "Preprocessor":
 
         return self._fit(dataset)
 
+    def update(self, dataset: Dataset) -> "Preprocessor":
```

Comment:
Two questions/concerns here:

1. Could we support incremental fitting without a new `.update()` API by letting `.fit()` accept `DatasetPipelines`?
2. This incremental fitting won't work if `.fit()` finalizes aggregations whose partial results are unmergeable, e.g. stddev for the standard scaler. If `.fit()` no longer finalizes aggregations, we'll then have to figure out when that finalization happens (should there be a `.finalize()` API?)

If we still want incremental fitting on plain Datasets, IMO it might be a cleaner API to have `.fit()` always finalize, and have a `.partial_fit()` API that's meant for incremental fitting, similar to sklearn, + a `.partial_finalize()` API to finalize/"seal" the fit.

Note that all of this can be handled under-the-hood of the `.fit()` API if we go with (1).